Interface.py

```diff
@@ -1,72 +1,195 @@
+                    wait = False
+
+        #if done waiting, or if not waiting, send a post to shut down operations
+        r = request.post(url = self.url_str + '/kill')
+        print('shut down achieved')
             
 
-#this is all about the references to the actual job object, a means to reference from client
-#the future object is responsible for getting inquiries from the client about the server
-#the future object should keep all the info for the server and name, and facilitate communication between server and client
+#future doesn't need much aside from a means to identify a job and the way to contact it in the server
+#when you read result, if its not completed it would just be blocked, as the job is still running
 class Future():
-    def __init__(self, job_name, server_info):
-       self.job_name = job_name
+    def __init__(self, fn, job_id, url_str, *args, **kwargs):
+       self.fn = fn
+       self.job_id = job_id
+       self.url_str = url_str
+       self.func_args = args
+       self.func_kwargs = kwargs
+    def future_info(self):
+        info_dict = {
+            'fn': self.fn, 
+            'job_id': self.job_id, 
+            'url_str': self.url_str, 
+        }
+        return info_dict
+        
     def cancel(self):
+        #check to see if it can be canceled, won't even attempt to cancel if it is running
+        if running():
+            #returns false if it is currently running
+            return False
+        else: 
+            r = requests.post(self.url_str + '/cancel-job', json = {'job_id': self.job_id})
+            #change this so that it can have info to cancel a job, and have it specific for a job_id
+            return True
 
-        return False
     def cancelled(self):
-        return True
+        r = requests.get(url = self.url_str + '/grab-job-info', json = {'job_id': self.job_id})
+        return r.json()['cancelled']
 
+    #distinct from the done method as this works whether started or not
     def running(self):
-        return True
+        #make sure you check to see if it was started, DO NOT depend on done()
+         r = requests.get(url = self.url_str + '/grab-job-info', json = {'job_id': self.job_id})
+         return r.json()['running']
+
     def done(self):
+         r = requests.get(url = self.url_str + '/grab-job-info', json = {'job_id': self.job_id})
+         if (r.json()['cancelled'] or r.json()['complete']):
+             return True
+         else:
+             return False
+
+    #can't do regular "done" thing, needs to 
+        #raise CancelledError
+        #and also mirror the exception raised by the call
+    def result(self, timeout = None):
+        time_check = True
+        return_value = None
+        timer = 0.0
+        while (time_check):
+            time.sleep(1.0)
+            timer += 1.0
+            r = requests.get(url = self.url_str + '/grab-job-info', json = {'job_id': self.job_id})
+            if (r.json['cancelled']):

```

Comment:
you perhaps better store the status in a single property, instead of using multiple booleans. Those status are mutually exclusive